middie.js

```diff
@@ -61,10 +62,16 @@ function middie (complete) {
       } else {
         if (!urls[i]) {
           functions[i](req, res, that.done)
-        } else if (urls[i] && (urls[i] === url || urls[i].indexOf(url) > -1)) {
-          functions[i](req, res, that.done)
         } else {
-          that.done()
+          if (urls[i] === url) {
+            functions[i](req, res, that.done)
+          } else if (typeof urls[i] !== 'string' && urls[i].indexOf(url) > -1) {
+            functions[i](req, res, that.done)
+          } else if (endOfPathAsterixRegex.test(urls[i]) && url.indexOf(urls[i].slice(0, -1)) > -1) {
```

Comment:
> this is a fairly expensive call, I think we should avoid this regexp check for every request, but rather do it once.

Don't know you want to implement that, do you want to store it on instance with something that say "this middleware use * in route" ?

> Regarding the indexOf and slice, we might have to write our own little functions that does a charcode-by-charcode comparison, as the slice call might be too expensive?

will dig into this later then, thanks for the review.